akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -369,4 +373,17 @@ object Source {
     new Source(new AcknowledgeSource(bufferSize, overflowStrategy, DefaultAttributes.acknowledgeSource, shape("AcknowledgeSource")))
   }
 
+  /**
+   * Creates a Source from a Files contents.
+   * Emitted elements are `chunkSize` sized [[akka.util.ByteString]] elements,
+   * except the final element, which will be up to `chunkSize` in size.
+   *
+   * You can configure the default dispatcher for this Source by changing the `akka.stream.blocking-io-dispatcher` or
+   * set it for a given Source by using [[ActorAttributes]].
+   *
+   * It materializes a [[Future]] containing the number of bytes read from the source file upon completion.
+   */
+  def file(f: File, chunkSize: Int = 8192): Source[ByteString, Future[Long]] =
```

Comment:
Smarter how? One would have to "explore different buffer sizes and try to find which one works", but on a machine which is impacted by various factors that may be a) nondeterministic b) too slow to figure out the best size for not too large streams.

Giving the option in the param at least allows people who want to tweak it to tweak it. Not sure if it's hurting to keep it - do you mean because "one more param => more complexity"? It's 2 params so... but that bad yet :)